app/models/post.rb

```diff
@@ -1,5 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :user
+  after_save :update_postscounter
+
+  belongs_to :author, class_name: 'User'
   has_many :comments
-  has_many :likes
+  has_many :posts
+
+  def update_posts_count
+    author.update(postscounter: author.posts.count)
+  end
```

Comment:
- [ ] Great job creating a method that updates the posts counter for a user. 👏🏻 👏🏻  However It would be nice to make the `update_post_count` method to be a `private` method 👍🏻 💯 Kindly make sure to add this to the remaining model as well ✍🏻 

- [ OPTIONAL]  🤔  You can also use the increment! method to make your code clean 👇🏻 :


   ```ruby
     # app/models/post.rb
     after_save :update_post_counter
   
     private
   
     def update_post_counter
       author.increment!(:posts_counter)
     end
   ```

- This is also application on the `# app/models/comment.rb` and `# app/models/like.rb` model

Reply:
Thank you. I will implement this right away.